File: core/src/com/unciv/ui/utils/GameSaver.java
package com.unciv.ui.utils;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.files.FileHandle;
import com.badlogic.gdx.utils.Json;
import com.unciv.ui.GameInfo;
import com.unciv.ui.UnCivGame;

public class GameSaver {
    public static final String saveFilesFolder = "SaveFiles";

    public static FileHandle GetSave(String GameName) {
        return Gdx.files.local(saveFilesFolder + "/" + GameName);
    }

    public static void SaveGame(UnCivGame game, String GameName) {
        GetSave(GameName).writeString(new Json().toJson(game.gameInfo), false);
    }

    public static void LoadGame(UnCivGame game, String GameName) {
        game.gameInfo = new Json().fromJson(GameInfo.class, GetSave(GameName).readString());
        game.gameInfo.setTransients();
    }
}

File: core/src/com/unciv/ui/utils/GameSaver.kt
package com.unciv.ui.utils

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.files.FileHandle
import com.badlogic.gdx.utils.Json
import com.unciv.logic.GameInfo

object GameSaver {
    private const val saveFilesFolder = "SaveFiles"

    fun getSave(GameName: String): FileHandle {
        return Gdx.files.local("$saveFilesFolder/$GameName")
    }

    fun getSaves(): List<String> {
        return Gdx.files.local(saveFilesFolder).list().map { it.name() }
    }

    fun saveGame(game: GameInfo, GameName: String) {
        getSave(GameName).writeString(Json().toJson(game), false)
    }

    fun loadGame(GameName: String) : GameInfo {
        val game = Json().fromJson(GameInfo::class.java, getSave(GameName).readString())
        game.setTransients()
        return game
    }

    fun deleteSave(GameName: String){
        getSave(GameName).delete()
    }
}
